tests/test_stdin_encoding_1426.py

```
@@ -0,0 +1,713 @@
+    `東京` is `e6 9d b1`, and `0x9d` is one of the five bytes cp1252 leaves
+    undefined, so an unpinned stdin fails at the read with "is not valid
+    UTF-8" and never reaches the session lookup.
+    """
+    payload = (
+        b'[{"index":0,"belief_type":"fact","persist":true,'
+        b'"note":"caf\xc3\xa9 \xe6\x9d\xb1\xe4\xba\xac"}]'
+    )
+    env = dict(os.environ)
+    env.update(CP1252)
+    env["HOME"] = str(tmp_path)
+    result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)